### ISSUE TITLE
Unexpected (positive!) issue: it works WITHOUT "synology_m2_volume" or "synology_enable_m2_volume"?

### ISSUE THREAD

**Issue by SurinameClubcard** [007revad/Synology_HDD_db#276]
I am using a Synology 1621+ with DSM 7.2.1-69057 update 4. I have inserted 2xSamsung NVMe 980 pro (1TB), which didn't show in the Storage Manager. So I ran the "synology_HDD_db" script once and rebooted. 

I expected to have to use either "synology_m2_volume" or "synology_enable_m2_volume" after that but to my surprise it seems that all functionality is available immediately without running any of those two scripts? 

Was that to be expected?

**Comment by 007revad** [comment]
Yes. It's pretty cool isn't it? :)

The DS1621+ is one of the 8 models that, since DSM 7.2, officially support M.2 volumes, so they only need synology_hdd_db.
https://kb.synology.com/en-global/DSM/tutorial/Which_models_support_M_2_SSD_storage_pool

**Comment by SurinameClubcard** [comment]
Thanks for that information! So it would also be okay to enable read/write cache on that?

I did notice one small issue. Apparently, I have two identical SATA SSDs in my Syno with two different firmwares. I didn't know that! The script showed me. If looks to me that the one with the older firmware is "overwritten" by the other?

Output below, where RVM01B6Q is not mentioned anymore in the JSON output:

```
bash-4.4# ./syno_hdd_db.sh -s
Synology_HDD_db v3.5.88
DS1621+ DSM 7.2.1-69057-4
StorageManager 1.0.0-0017

ds1621+_host_v7 version 8051

Using options: -s
Running from: [redacted]

HDD/SSD models found: 3
SSD 860 PRO 512GB,RVM01B6Q
SSD 860 PRO 512GB,RVM02B6Q
WD8003FFBX-68B9AN0,83.00A83

M.2 drive models found: 1
Samsung SSD 980 PRO 1TB,5B2QGXA7

No M.2 PCIe cards found

No Expansion Units found

Added SSD 860 PRO 512GB to ds1621+_host_v7.db
Edited unverified drives in ds1621+_host_v7.db
Updated SSD 860 PRO 512GB in ds1621+_host_v7.db
WD8003FFBX-68B9AN0 already exists in ds1621+_host_v7.db
Added Samsung SSD 980 PRO 1TB to ds1621+_host_v7.db

Backed up synoinfo.conf

Support disk compatibility already enabled.

Support memory compatibility already enabled.

NVMe support already enabled.

M.2 volume support already enabled.

Drive db auto updates already enabled.

    "SSD 860 PRO 512GB": {
      "RVM02B6Q": {
        "compatibility_interval": [
          {
            "compatibility": "support",
            "not_yet_rolling_status": "support",
            "fw_dsm_update_status_notify": false,
            "barebone_installable": true,
            "smart_test_ignore": false,
            "smart_attr_ignore": false
          }
        ]

    "SSD 860 PRO 512GB": {
      "RVM02B6Q": {
        "compatibility_interval": [
          {
            "compatibility": "support",
            "not_yet_rolling_status": "support",
            "fw_dsm_update_status_notify": false,
            "barebone_installable": true,
            "smart_test_ignore": false,
            "smart_attr_ignore": false
          }
        ]

    "WD8003FFBX-68B9AN0": {
      "83.00A83": {
        "size_gb": 8000,
        "compatibility_interval": [
          {
            "compatibility": "support",
            "not_yet_rolling_status": "support",
            "fw_dsm_update_status_notify": false,
            "barebone_installable": true,
            "smart_test_ignore": false,
            "smart_attr_ignore": false
          }

    "Samsung SSD 980 PRO 1TB": {
      "5B2QGXA7": {
        "compatibility_interval": [
          {
            "compatibility": "support",
            "not_yet_rolling_status": "support",
            "fw_dsm_update_status_notify": false,
            "barebone_installable": true,
            "smart_test_ignore": false,
            "smart_attr_ignore": false
          }
        ]

DSM successfully checked disk compatibility.

You may need to reboot the Synology to see the changes.
bash-4.4#
```

**Comment by 007revad** [comment]
> Thanks for that information! So it would also be okay to enable read/write cache on that?
Yes.
 
> I did notice one small issue. Apparently, I have two identical SATA SSDs in my Syno with two different firmwares. I didn't know that! The script showed me. If looks to me that the one with the older firmware is "overwritten" by the other?
> 
> Output below, where RVM01B6Q is not mentioned anymore in the JSON output:
> 
> ```
>     "SSD 860 PRO 512GB": {
>       "RVM02B6Q": {
>         "compatibility_interval": [
>           {
>             "compatibility": "support",
>             "not_yet_rolling_status": "support",
>             "fw_dsm_update_status_notify": false,
>             "barebone_installable": true,
>             "smart_test_ignore": false,
>             "smart_attr_ignore": false
>           }
>         ]
> 
>     "SSD 860 PRO 512GB": {
>       "RVM02B6Q": {
>         "compatibility_interval": [
>           {
>             "compatibility": "support",
>             "not_yet_rolling_status": "support",
>             "fw_dsm_update_status_notify": false,
>             "barebone_installable": true,
>             "smart_test_ignore": false,
>             "smart_attr_ignore": false
>           }
>         ]
> ```

You've found a bug in script's code for displaying the database changes.

This command should correctly show you the SSD 860 PRO 512GB including both firmware versions
```
jq . /var/lib/disk-compatibility/ds1621+_host_v7.db | grep -A 31 "SSD 860 PRO 512GB"
```

**Comment by 007revad** [comment]
Fixed in v3.5.89

https://github.com/007revad/Synology_HDD_db/releases

**Comment by SurinameClubcard** [comment]
Thanks again for your efforts! Very much appreciated!!! Closing this (initially) (non-)issue.
